library/core/src/slice/ascii.rs

```diff
@@ -80,6 +80,98 @@ impl [u8] {
     pub fn escape_ascii(&self) -> EscapeAscii<'_> {
         EscapeAscii { inner: self.iter().flat_map(EscapeByte) }
     }
+
+    /// Returns a byte slice with leading ASCII whitespace removed.
+    ///
+    /// 'Whitespace' refers to the following ASCII characters:
+    ///
+    /// - Null (0x00)
+    /// - Horizontal tab \t (0x09)
+    /// - Line feed \n (0x0A)
+    /// - Vertical tab \v (0x0B)
+    /// - Form feed \f (0x0C)
+    /// - Carriage return \r (0x0D)
+    /// - Space (0x20)
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(byte_slice_trim_ascii)]
+    ///
+    /// assert_eq!(b" \t hello world\n".trim_start_ascii(), b"hello world\n");
+    /// assert_eq!(b" ".trim_start_ascii(), b"");
+    /// ```
+    #[unstable(feature = "byte_slice_trim_ascii", issue = "none")]
+    pub fn trim_start_ascii(&self) -> &[u8] {
+        let mut index = 0;
+        for i in 0..self.len() {
+            match self[i] {
+                0x00 | 0x09 | 0x0A | 0x0B | 0x0C | 0x0D | 0x20 => index += 1,
```

Comment:
Seems like [u8::is_ascii_whitespace](https://doc.rust-lang.org/nightly/std/primitive.u8.html#method.is_ascii_whitespace) is probably the right thing to call here, rather than a manual listing?

Reply:
Oh, perfect!